### PR TITLE
Wrapper update

### PR DESCRIPTION
**Pull request by helomri**
Hello,
This little pull request is to update the [dependencies](https://github.com/EduWireApps/kdecole-api-dart/commit/a666684885fb910c998b51b91be17eeab792de91) and to make the code more simple to use; by example there is a new field in some entities to know if the object has all it's content loaded from the ENT. I also fixed a bug where if a post on any ENT is any other type than a "actualite" the URL to fetch it's content is different. This is why the wrapper now use the "type" field to know what will be the URL to fetch for the post.

### CODE REVIEW

**Review by Freezman31 — CHANGES_REQUESTED** [review]
Very good pr !

File: lib/src/client.dart [hunk 1]
```diff
@@ -210,30 +219,31 @@ class Client {
...
 
   ///Mark as read an Email
-  Future<void> markAsRead(Email email) async {
-    await _invokeApi('messagerie/communication/lu/${email.id}/', 'PUT');
+  void markAsRead(Email email) {
+    _invokeApi('messagerie/communication/lu/${email.id}/', 'PUT');
   }
 
   ///Delete communication
-  Future<void> deleteMail(Email email) async {
```
[the remark this answers is not in the document]
Comment (theskyblockman) [reply lib/src/client.dart:229]: We don't use the response of the request so we do not need to await it

File: lib/src/entities/email.dart [hunk 2]
```diff
@@ -16,5 +17,6 @@ class Email {
       required this.receivers,
       required this.id,
       required this.messages,
-      required this.read});
+      required this.read,
+      required this.isComplete});
```
[the remark this answers is not in the document]
Comment (theskyblockman) [reply lib/src/entities/email.dart]: Changed the fields to isComplete to be more explicit in [this](https://github.com/EduWireApps/kdecole-api-dart/pull/9/commits/efd0e2e78b6ac4434d937bc0722cbaafe85043fa) commit

File: lib/src/client.dart [hunk 3]
```diff
@@ -210,30 +219,31 @@ class Client {
...
```
Comment (Freezman31) [inline lib/src/client.dart:229]: Same problem ?

**Review by Freezman31 — APPROVED** [review]
Looks good !
Did you test it ?

**Comment by theskyblockman** [comment]
Yes I tried it with a local server which reproduce the requests and all seem good

**Pull request merged**